spec/services/renewal_letters_bulk_pdf_service_spec.rb

```diff
@@ -0,0 +1,31 @@
+# frozen_string_literal: true
+
+require "rails_helper"
+
+RSpec.describe RenewalLettersBulkPdfService do
+  describe ".run" do
+    let(:registrations) { create_list(:registration, 2) }
+    let(:result) { double(:result) }
+
+    it "returns a PDF string with a letter per-registration" do
+      expect(Airbrake).to_not receive(:notify)
+
+      expect(described_class.run(registrations)).to start_with("%PDF")
```

Comment:
This is the same expectation for the test `when there are no registrations`, which for me feels like we're not really testing the 2 contexts.

It'd be great if there was something else we could check that confirms there is a letter per registration, else I'd just update the test statement

```ruby
      it "returns a PDF" do
```

Reply:
We can't do that without adding the ability to read PDF content 🤔 shall I research that? I can probably bring in Prawn in the `:test` group only.

Reply:
I'd lean towards

- just checking we have a PDF for the first test (there are registrations)
- changing the design and returning `nil` (when there are no registrations)

On that last point it kind of makes more sense to me. If there are no registrations I would not expect to generate a PDF, rather than generate a blank/empty one.